src/plots/font_attributes.js

```diff
@@ -99,6 +99,42 @@ module.exports = function(opts) {
             ].join(' ')
         },
 
+        textcase: opts.noFontTextcase ? undefined : {
+            editType: editType,
+            valType: 'enumerated',
+            values: ['normal', 'word caps', 'upper', 'lower'],
```

Comment:
I saw these values `'normal'`, `'word caps'`, etc. appear earlier - should they be defined somewhere as constants or is that just adding useless indirection?

Reply:
Good point.
We could declare these keys in a separate file as constant and used them here as well as in 
[src/components/drawing/index.js](https://github.com/plotly/plotly.js/pull/6983/files#diff-e8c93e83ebf0c3a486f3187d44d91b1709f51bccaba559fcadab2a3e3f9f56c6).
But it's easier to read them here and possibly provide a description if needed.

BTW I totally agree with you that to improve performance we could/should declare these constant arrays as constants in higher scope i.e. in several places in our source code (and possibly after switching to ES6).